src/core/qgsvectorlayer.cpp

```diff
@@ -2387,6 +2387,19 @@ long QgsVectorLayer::featureCount() const
          ( mEditBuffer ? mEditBuffer->mAddedFeatures.size() - mEditBuffer->mDeletedFeatureIds.size() : 0 );
 }
 
+long QgsVectorLayer::featureCount( QgsFeatureRequest request )
```

Comment:
I'm interesting in to know the number of features for e filtered set because in WFS 2.0, I have to return the number in the Document element before I fetched all or only a part of the features.
In WFS 2.0, a user can request 10 features with a filter and an offset. To help software and user to know the number of request to do to have all the features, the Document element contains the number of features for a filtered set.

That said, I'll close this Pull-Request and will work on adding count to feature iterator to use compiled expressions.

Thanks for your reply.